app/controllers/course/sections_controller.rb

```diff
@@ -1,7 +1,11 @@
 class Course::SectionsController < Course::ApplicationController
   def show
     @section = Section.find(params[:id])
-    @quiz_session = get_quiz.session
+    if @section.questions.count == 0
+      @quiz_session = nil
+    else
+      @quiz_session = get_quiz.session
+    end
```

Comment:
I get the logic, but it feels better to leave the controller as it was and do something like this:
```ruby
# quiz.rb
def session
  Quiz::Session.new(quiz: self) if questions.any?
end
```
This way the quiz is responsible for returning the relevant session according to its internal state (which I'd rather avoid checking in the controller). In the future we can create a null session for quizzes if it becomes needed, this gets us closer to it and leave the controller cleaner of business logic.